components/dashboard/src/components/PageWithSubMenu.tsx

```diff
@@ -20,30 +23,64 @@ export interface PageWithSubMenuProps {
-    const location = useLocation();
     return (
         <div className="w-full">
             <Header title={p.title} subtitle={p.subtitle} tabs={p.tabs} />
-            <div className="app-container flex pt-9">
+            <div className="app-container flex md:pt-9 flex-col md:flex-row">
+                {/* TODO: extract into SubMenu component and show scrolling indicators */}
                 <div>
-                    <ul className="flex flex-col text tracking-wide text-gray-500 pt-4 lg:pt-0 w-52 space-y-2">
+                    <ul
+                        className={classNames(
+                            // Handle flipping between row and column layout
+                            "flex flex-row md:flex-col items-center",
+                            "w-full md:w-52 overflow-auto md:overflow-visible",
+                            "pt-4 pb-4 md:pb-0",
+                            "space-x-2 md:space-x-0 md:space-y-2",
+                            "tracking-wide text-gray-500",
+                        )}
+                    >
                         {p.subMenu.map((e) => {
-                            let classes = "flex block py-2 px-4 rounded-md";
-                            if (e.link.some((l) => l === location.pathname)) {
-                                classes += " bg-gray-300 text-gray-800 dark:bg-gray-800 dark:text-gray-50";
-                            } else {
-                                classes += " hover:bg-gray-100 dark:hover:bg-gray-800";
-                            }
-                            return (
-                                <Link to={e.link[0]} key={e.title}>
-                                    <li className={classes}>{e.title}</li>
-                                </Link>
-                            );
+                            return <SubmenuItem title={e.title} link={e.link} key={e.title} />;
                         })}
                     </ul>
                 </div>
-                <div className="ml-16 lg:ml-32 w-full pt-1 mb-40">{p.children}</div>
+                <div className="md:ml-16 lg:ml-32 w-full pt-1 mb-40">
+                    <Separator className="md:hidden" />
+                    <div className="pt-4 md:pt-0">{p.children}</div>
+                </div>
             </div>
         </div>
     );
 }
+
+type SubmenuItemProps = {
+    title: string;
+    link: string[];
+};
+
+const SubmenuItem: FC<SubmenuItemProps> = ({ title, link }) => {
+    const location = useLocation();
+    const itemRef = useRef<HTMLLIElement>(null);
+
+    // TODO: can remove this once we use sub-routes and don't re-render the whole page for each sub-route
+    useEffect(() => {
+        if (itemRef.current && link.some((l) => l === location.pathname)) {
+            itemRef.current.scrollIntoView({ behavior: "auto", block: "nearest", inline: "start" });
+        }
+    }, [link, location.pathname]);
+
+    let classes = "flex block py-2 px-4 rounded-md whitespace-nowrap max-w-52";
```

Comment:
**suggestion**: Probably not the best line to comment on this, but the initial idea waw to reuse the same tabs pattern we use in the dashboard. It should be ok to leave is as is for this PR but it'd be nicer for users to face a pattern they've seen before. Thoughts?

| BEFORE | AFTER |
|-|-|
| ![bmh-respon4a5450a2ee preview gitpod-dev com_usage(Pixel 5)](https://user-images.githubusercontent.com/120486/227504893-2e5be324-1b40-4fca-8e88-25ecced67ef4.png) | ![bmh-respon4a5450a2ee preview gitpod-dev com_usage(Pixel 5) (1)](https://user-images.githubusercontent.com/120486/227504897-953b809d-d752-4b52-a7b5-f888a7eddcdd.png) |